mkosi/machine.py

```diff
@@ -122,9 +146,50 @@ def run(self, commands: Sequence[str], timeout: int = 900, check: bool = True) -
 
         return process
 
+    def kill(self, *args: Any, **kwargs: Any) -> None:
+        self.__exit__(*args, **kwargs)
```

Comment:
So I'd go ahead and remove the arguments from kill() just like we removed them from reboot() and pass None, None, None to `__exit__()` instead. We don't really use the exception information anyway so no need to complicate the signature of kill() so we can pass along exception information we don't use.

This would also mean we can get rid of `exec_info` in `MkosiMachineTest` and we wouldn't need to override `_callTestMethod` anymore either, which would be a nice simplification.